cci_nepal/analysis/dummy_data/dummy_data_modeling/model_analysis.py

```diff
@@ -0,0 +1,295 @@
+
+
+df_dummy.head(5)
+
+# This part is just to test if all 3 labels are predicted well in the model ###
+
+# Since most of the predictions were otherwise averaged around 2 ###
+
+
+def replace_with_number(df, columns, values):
+    for one_column, one_value in zip(columns, values):
+        df[one_column] = one_value
+
+    return df
+
+
+replace_with_number(df_dummy, ["Blanket", "Printed Cloth", "Hand Sanitizer"], [3, 1, 3])
+
+
+# Columns to categorize for Linear Regression modeling #
+
+num_cols = df_dummy._get_numeric_data().columns
+
+columns_to_categorize = list(set(df_dummy.columns) - set(num_cols))
+
+df_dummy[columns_to_categorize] = df_dummy[columns_to_categorize].astype("category")
+
+
+model_basic = LinearRegression()
+model_non_basic = LinearRegression()
+
+
+df_dummy.shape
+df_dummy.iloc[:, 0:33]
+
+# Features list to be selected here #
+
+X = df_dummy.iloc[:, 0:33]  # All features fed in for now
+X = pd.get_dummies(data=X, drop_first=True)
+logging.info(X.columns)
+X
+
+
+nfri_basic = [
+    "Plastic Tarpaulin",
+    "Blanket",
+    "Sari",
+    "Male Dhoti",
+    "Shouting Cloth / Jeans",
+    "Printed Cloth",
+    "Terry Cloth",
+    "Utensil Set",
+    "Water Bucket",
+    "Nylon Rope",
+    "Sack (Packing Bag)",
+]
+
+nfri_non_basic = [
+    "Cotton Towel",
+    "Laundry Soap",
+    "Tooth Brush and Paste",
+    "Sanitary Pad",
+    "Ladies Underwear",
+    "Torch Light",
+    "Whistle Blow",
+    "Nail Cutter",
+    "Hand Sanitizer",
+    "Liquid Chlorine",
+]
+
+y_basic = df_dummy.loc[:, nfri_basic]
+y_basic
+
+y_non_basic = df_dummy.loc[:, nfri_non_basic]
+y_non_basic
+
+X_train_basic, X_test_basic, y_train_basic, y_test_basic = train_test_split(X, y_basic)
+(
+    X_train_non_basic,
+    X_test_non_basic,
+    y_train_non_basic,
+    y_test_non_basic,
+) = train_test_split(X, y_non_basic)
+
+
+model_basic.fit(X_train_basic, y_train_basic)
+model_non_basic.fit(X_train_non_basic, y_train_non_basic)
+
+
+test_predictions_basic = model_basic.predict(X_test_basic)
+test_predictions_non_basic = model_non_basic.predict(X_test_basic)
+
+test_predictions_labels_basic = [
+    [numeric_score_transformer(i) for i in nested] for nested in test_predictions_basic
+]
+test_predictions_items_wise_basic = list(
+    map(list, zip(*test_predictions_labels_basic))
+)  # Transforming the above list
+
+test_predictions_labels_non_basic = [
+    [numeric_score_transformer(i) for i in nested]
+    for nested in test_predictions_non_basic
+]
+test_predictions_items_wise_non_basic = list(
+    map(list, zip(*test_predictions_labels_non_basic))
+)  # Transforming the above list
+
+
+accuracy_list_basic = []
+for i in range(0, len(nfri_basic)):
+    accuracy_list_basic.append(
+        accuracy_score(test_predictions_items_wise_basic[i], y_test_basic.iloc[:, i])
+    )
+
+logging.info(
+    pd.DataFrame(accuracy_list_basic, y_test_basic.columns, columns=["Accuracy"])
+)
+
+
+accuracy_list_non_basic = []
```

Comment:
As we are looking at the accuracy of the categories, it would be useful to see some of these in the form of a confusion matrix? So again, looking at 'Plastic Tarpaulin' to know what are the actual answers compared to the predicted for essential, desirable and unnecessary. Part of the reason for this is to see if the model is developing certain behaviour that will give it the best score (like just predicting everything as desirable if thats the majority class).